src/prefect/cli/execute.py

```diff
@@ -74,7 +76,10 @@ def cloud_flow():
     except Exception as exc:
         msg = "Failed to load and execute Flow's environment: {}".format(repr(exc))
         state = prefect.engine.state.Failed(message=msg)
-        version = result.data.flow_run[0].version
-        client.set_flow_run_state(flow_run_id=flow_run_id, version=version, state=state)
+        if flow_run:
+            version = result.data.flow_run[0].version
+            client.set_flow_run_state(
```

Comment:
If the client initialization fails, then `client` won't be defined and this will raise an error, putting you back where you started